libs/block-mesh-manager/src/ws/connection_manager.rs

```diff
@@ -33,124 +34,79 @@ impl ConnectionManager {
         }
     }
 
-    pub fn cron_reports(&self, period: Duration) -> JoinHandle<()> {
-        let broadcaster = self.broadcaster.clone();
-        tokio::spawn(async move {
-            loop {
-                broadcaster
-                    .queue_multiple(
-                        &[
-                            WsServerMessage::RequestUptimeReport,
-                            WsServerMessage::RequestBandwidthReport,
-                        ],
-                        100,
-                    )
-                    .await;
-                tokio::time::sleep(period).await;
-            }
-        })
+    pub async fn cron_reports(
+        &mut self,
+        period: Duration,
+        messages: impl Into<Vec<WsServerMessage>> + Clone + Send + 'static,
+        window_size: usize,
+        pool: PgPool,
+    ) -> anyhow::Result<CronReportsController> {
+        self.broadcaster
+            .cron_reports(period, messages, window_size, pool)
+            .await
     }
 }
-#[derive(Debug, Clone)]
-pub struct Broadcaster {
-    global_transmitter: broadcast::Sender<WsServerMessage>,
-    sockets: Arc<DashMap<(Uuid, SocketAddr), mpsc::Sender<WsServerMessage>>>,
-    queue: Arc<Mutex<VecDeque<(Uuid, SocketAddr)>>>,
-}
 
-impl Broadcaster {
-    fn new() -> Self {
-        let (global_transmitter, _) = broadcast::channel(10000);
-        Self {
-            global_transmitter,
-            sockets: Arc::new(DashMap::new()),
-            queue: Arc::new(Mutex::new(VecDeque::new())),
+pub async fn settings_loop(
+    pool: &PgPool,
+    user_id: &Uuid,
+    period: Duration,
+    messages: impl Into<Vec<WsServerMessage>> + Clone + Send + 'static,
+    window_size: usize,
+    broadcaster: Broadcaster,
+    stats_tx: Sender<CronReportStats>,
```

Comment:
Someone is listening on the RX side? It seems to be in main.rs and the value isn't used.